sentinelsat/sentinel.py

```diff
@@ -939,24 +885,51 @@ def check_bounds(geometry):
     wkt = re.sub(r'(?<!\d) ', '', wkt)
     return wkt
 
-
+def placename_to_wkt(placename):
+    """Maps the placename to rectangular bounding extents using Nominatim API and converts to Well-Known-Text.
+    Parameters
+    ----------
+    placename : str
+	the query to geocode
+        
+    Returns
+    -------
+    polygon coordinates
+        string of comma separated coordinate tuples (lon, lat) to be used by SentinelAPI
+    """
+    rqst = requests.post('https://nominatim.openstreetmap.org/search', \
+                         params = { 'q':placename,'format':'geojson'})
+    #Check that the response from Openstreetmapserver has status code 2xx 
+    #and that the response is valid JSON.
+    _check_scihub_response(rqst)
+    try:
+        jsonlist=rqst.json()
+        #Get the First result's bounding box.
+        bbox= jsonlist['features'][0]['bbox']
+        #the bbox follows the pattern:
+        #[longitude lowerleft corner,latitude lowerleft corner,longitude upperright corner,latitude upperright corner]
+        footprint="POLYGON(({} {},{} {},{} {},{} {},{} {}))"
+        footprint=footprint.format(bbox[0],bbox[3],bbox[2],bbox[3], \
+                                   bbox[2],bbox[1],bbox[0],bbox[1], \
+                                   bbox[0],bbox[3])
+        return footprint
+    except:
```

Comment:
Please try to catch a specific exception. With this statement you are blanket catching *all* exceptions and returning an IndexError, even though the source error might be something else.

Reply:
Yes. Which line can cause which exception? Wrap only that line in a `try ... except KeyError:` or `try ... except IndexError:` or what else it might be. Then you can me much more sure that the exception you catch is the one you intend to (and not something totally different that you misinterpret).